dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -205,7 +205,7 @@ export const TableWithFavorite = () => {
                     </Td>
                     <Td dataLabel={columnNames.name}>{repo.name}</Td>
                     <Td dataLabel={columnNames.creationDate}>
-                      {repo.metadata["dataset.created"]}
+                      {moment(repo.metadata["dataset.created"]).format("YYYY-MM-DDTHH:mm")}
```

Comment:
It looks to me ([e.g.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toUTCString)) like Javascript already has support for converting between both ISO and JSON format date strings and `Date`'s.  What else do we need `moment` or Luxon for?

Reply:
OK, one important question here!!

Javascript do support converting to ISO String (`date.toISOString()`)

But the problem with this is we cannot pass any parameters to toISOString() function, so if we want our dates in a certain way, we would have to resort to string operations to get the work done.That is when libraries like moment or luxon comes to picture!!

For Instance, the same work above , if to be done by using Javascript Date methods,would look like something like this

`repo.metadata["dataset.created"].toISOString().slice(0,16)`

Since toISOString() would always return the date in`YYYY-MM-DDTHH:mm:ss.000Z` format

So now, it boils down to which method looks more cleaner and more maintainable going further

Reply:
We want to be able to move towards localization based on browser or user-specific settings, so we definitely will want to be able to handle more than just straight ISO-8601 formatting. What we use internally for comparisons is a different matter, although I'd tend towards using only `Date` objects internally rather than comparing strings.

Again, if we want to do "day" comparisons, there are date operators (either through `moment` or `luxon`, if not `Date`) to determine the beginning of day and end of day for a given `Date` object rather than dropping the time from a string date.

Reply:
First I'll say, given that we have greater aspirations with respect to handling dates (e.g., handling localization), it would be best to represent all dates internally as `Date` objects (and then we can defer the details of displaying them); this will, among other things, make it easy to compare them to each other, without resorting to things like `moment` or formatting.

That said, I brought up string comparisons because I believe that `repo.metadata["dataset.created"]` is a _string_.  (Please correct me if I'm wrong....)

Javascript apparently has some interesting dynamic type conversions which allow `Date` objects and strings to be compared.  I would not be inclined to rely on those.  So, prior to doing a comparison, I think we should explicitly convert both operands to the same type.  I think we now agree that that type should be a `Date`, but I was contemplating using strings, because, in most of the cases in this review, the operands were already strings, and it seem overworked to convert them all to `Date`'s just to compare them.  However, if we were to go over to storing them as `Date` objects, then the comparisons would be easy, and we just need to deal with conversions when they are displayed to or set by the user.

Reply:
> It looks to me ([e.g.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toUTCString)) like Javascript already has support for converting between both ISO and JSON format date strings and `Date`'s. What else do we need `moment` or Luxon for?

It seems we can't use .toISOString here,since repo.metadata["dataset.created"] is a string and the .toISOString can only be applied to date objects

Reply:
We presumably _could_ do this:
```
                      {Date(repo.metadata["dataset.created"]).toISOString()}
```

But, what I think we want to do is, when we receive the response from the server, convert any dates in it to Javascript `Date` objects, so that they will already be `Date`'s when it's time to do stuff with them.

Reply:
I have not given much thought about it, but my basic understanding says that if we want to convert all the dates recieved from the server to Javascript dates ,we would have to loop the dataset array recieved which might add additional time complexity (It might be not that much,may be negligible,but still it seems unnecessary on my initial thought).

Reply:
Yes, there is potentially a trade-off point.  If we're getting a lot of dates from the server but using only a few, then it would be better to defer converting them until they are needed.  However, this means that the code has to be aware of which "kind" of date it has in hand, and that makes the code harder to maintain.  Instead, if we typically end up using (in one way or another) nearly every date that we get from the server, then we should loop over the dataset when it is received and convert them all.  (If we're taking the response and converting it to another form already, that would be a good point at which to rework all the dates; otherwise, potentially, we could convert them "in place", replacing the JSON string values with JS `Date` object references.)